pkg/orchestrator/data_scrubber.go

```diff
@@ -0,0 +1,125 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build orchestrator
+
+package orchestrator
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+)
+
+var (
+	defaultSensitiveWords = []string{
+		"password", "passwd", "mysql_pwd",
+		"access_token", "auth_token",
+		"api_key", "apikey", "pwd",
+		"secret", "credentials", "stripetoken"}
+)
+
+// DataScrubber allows the agent to block cmdline arguments that match
+// a list of predefined and custom words
+type DataScrubber struct {
+	Enabled                  bool
+	RegexSensitivePatterns   []*regexp.Regexp
+	LiteralSensitivePatterns []string
+	scrubbedCmdlines         map[string][]string
+}
+
+// NewDefaultDataScrubber creates a DataScrubber with the default behavior: enabled
+// and matching the default sensitive words
+func NewDefaultDataScrubber() *DataScrubber {
+	newDataScrubber := &DataScrubber{
+		Enabled:                  true,
+		LiteralSensitivePatterns: defaultSensitiveWords,
+		scrubbedCmdlines:         make(map[string][]string),
+	}
+
+	return newDataScrubber
+}
+func (ds *DataScrubber) ContainsSensitiveWord(word string) bool {
+	for _, pattern := range ds.LiteralSensitivePatterns {
```

Comment:
This part can potentially just use a simple LRU cache as there is no need to check for Contains everytime. 
Additionally, these are simple word strings. Wdyt? @hkaj

Reply:
what would the lru cache help with? I'm not sure i understand

Reply:
The idea was in case we want to stick with regex scrubbing we could still use a memory bounded e.g. LRU cache.
Each time we match a word with the regex we can save the word in the cache and safe performance.  

Why env? Commands are difficult, as they are a lot of different ones and we have to choose a good key (maybe just the command itself?).
But caching Envs should be easy as they are only 1 word. Does this make more sense? 💭